sdk/core/core-rest-pipeline/src/fetchHttpClient.ts

```diff
@@ -273,6 +277,7 @@ function buildBodyStream(
         const { done, value } = await reader.read();
         // When no more data needs to be consumed, break the reading
         if (done || !value) {
+          onEnd?.();
```

Comment:
do we need to worry about implementing `cancel()` in addition to `pull()`?

Reply:
I don't think we need to implement `cancel()` functionality for the Abort Controller. That is an additional feature that might be out of scope for this PR

Reply:
Yeah, I'm just curious if the consumer of the stream calls `cancel()` on it, will we clean up this listener or will it leak?

Looking at the docs here: https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/ReadableStream#parameters